manifests/updatesd.pp

```diff
@@ -1,15 +1,30 @@
-# == Class: yum::updatesd
+# @summary Manage yum-updatesd which is available on EL5, though not EL6
+#   This is defaulted to being disabled.
 #
-# Manage yum-updatesd which is available on EL5, though not EL6
+# @param updatesd_package
+#   Specify name of yum updatesd package.
 #
-# This is defaulted to being disabled.
+# @param updatesd_package_ensure
+#   Set ensure attribute for package resource. Valid values are: 'present',
+#   'absent', 'latest', and 'purged'.
+#
+# @param updatesd_service
+#   Specify name of yum updatesd service.
+#
+# @param updatesd_service_ensure
+#  Set ensure attribute for service resource. Valid values are: 'running'
+#  and 'stopped'.
+#
+# @param updatesd_service_enable
+#  Set enable attribute for service resource. Valid values are: 'true', 'false',
+#  'manual', and 'mark'.
 #
 class yum::updatesd (
-  $updatesd_package        = 'yum-updatesd',
-  $updatesd_package_ensure = 'absent',
-  $updatesd_service        = 'yum-updatesd',
-  $updatesd_service_ensure = 'stopped',
-  $updatesd_service_enable = false,
+  String $updatesd_package                                               = 'yum-updatesd',
+  Enum['absent', 'latest', 'present', 'purged'] $updatesd_package_ensure = 'absent',
+  String $updatesd_service                                               = 'yum-updatesd',
+  Enum['running', 'stopped'] $updatesd_service_ensure                    = 'stopped',
+  Enum['false', 'manual', 'mark', 'true'] $updatesd_service_enable       = 'false', # lint:ignore:quoted_booleans
```

Comment:
Perhaps instead of us doing type validation for enable and ensure, we should just check that a string or boolean is being passed. Then if Puppet adds another possible value for `enable` or whatever, it would be supported. If you pass the wrong thing, Puppet will already error.

Reply:
For ensure on services we can use [Stdlib::Ensure::Service](https://github.com/puppetlabs/puppetlabs-stdlib#stdlibensureservice).